### PR TITLE
Flying mobs now clip past non flying mobs, various flightsuit/iontrail/movement tweaks/code improvements/bugfixes

### PR DESCRIPTION
**Pull request by silicons**
:cl:
experimental: Flying mobs now fly over those that aren't flying.
experimental: Flightsuits will now be unstable if EMPed! Don't get ion'd! 
bugfix: Flightsuits now allow proper pulling
experimental: Moved now has an argument for if it was a regular Move or a forceMove.
rscadd: Instead of dumb sleep()s, follow trails now use SSfastprocess for processing!
rscadd: Mobs now float while flying automatically!
rscdel: Flightsuits can no longer be safely shut off if you're still barreling down the hallway.
experimental: Flightsuits should be less shitcode. Hope this PR doesn't blow anything important up!
/:cl:
removed crash damage from flightsuits the health loss should be enough.

### CODE REVIEW

**Comment by PraiseRatvar** [comment]
> experimental: Flightsuits will now be unstable if EMPed! Don't get ion'd!

explain

**Comment by kevinz000** [comment]
emp = each move has a chance to throw you off by one tile instead of being perfectly accurate.
pulling = fixes pulling being interrupted when you fly past things.

**Comment by DoctoreMad** [comment]
Does crashing into a wall due to EMP damage you?

**Comment by kevinz000** [comment]
hmmm i really should make emps more lethal

File: code/modules/clothing/spacesuits/flightsuit.dm [hunk 1]
```diff
@@ -13,19 +13,18 @@
...
-	var/item_state_boost = "flightpack_boost"
 	actions_types = list(/datum/action/item_action/flightpack/toggle_flight, /datum/action/item_action/flightpack/engage_boosters, /datum/action/item_action/flightpack/toggle_stabilizers, /datum/action/item_action/flightpack/change_power, /datum/action/item_action/flightpack/toggle_airbrake)
 	armor = list(melee = 20, bullet = 20, laser = 20, energy = 10, bomb = 30, bio = 100, rad = 75, fire = 100, acid = 75)
-
 	w_class = WEIGHT_CLASS_BULKY
 	slot_flags = SLOT_BACK
 	resistance_flags = FIRE_PROOF
 
-	var/processing_mode = FLIGHTSUIT_PROCESSING_FULL
+	var/icon_state_active = "flightpack_on"
```
Comment (optimumtact) [inline code/modules/clothing/spacesuits/flightsuit.dm]: why are these vars and not defines?

Comment (optimumtact) [reply code/modules/clothing/spacesuits/flightsuit.dm]: vars have a performance impact

File: code/modules/mob/living/living.dm [hunk 2]
```diff
@@ -620,6 +641,7 @@
 	if(!override)
 		float(!has_gravity)
 
+//TODO: Better floating
```
Comment (optimumtact) [inline code/modules/mob/living/living.dm]: can you not add these? they very rarely get followed up on, keep a todo list in your own management tool of choice

File: code/modules/clothing/spacesuits/flightsuit.dm (hunk 1 above)
Comment (kevinz000) [reply code/modules/clothing/spacesuits/flightsuit.dm]: is varediting an acceptable response?

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -120,7 +118,31 @@
 			ContactContractDisease(D)
 
 	if(now_pushing)
-		return 1
+		return TRUE
+
+	//TODO FOR LATER PRS: Make passing tables an automatic thing for flying and passable objects be determined better to prevent huge amounts of flags being set when mobs fly.
+	if(((movement_type & FLYING) && !(M.movement_type & FLYING)) || (!(movement_type & FLYING) && (M.movement_type & FLYING)))	//Fly past each other.
```
Comment (duncathan) [inline code/modules/mob/living/living.dm]: `if((movement_type & FLYING) ^ (M.movement_type & FLYING))`

**Review by duncathan — CHANGES_REQUESTED** [review]

**Review by duncathan — APPROVED** [review]

**Pull request merged**